managementAgent.js

```diff
@@ -1,51 +1,130 @@
+        process.on('SIGTERM', this.stop.bind(this));
+        process.on('SIGPIPE', () => {});
+    }
+
+    start() {
+        /* Define REPORT_TOKEN env variable needed by the management
+         * module. */
+        process.env.REPORT_TOKEN = process.env.REPORT_TOKEN
+          || _config.reportToken
+          || Uuid.v4();
+
+        metadata.setup(error => {
+            if (error) {
+                const msg = `failed to setup metadata: ${error}`;
+                throw new Error(msg);
+            }
+            initManagement(logger.newRequestLogger(), (overlay) => {
+                if (overlay) {
+                    this.loadedOverlay = overlay;
+                    this.startServer();
+                } else {
+                    const msg = `failed to init management: ${error}`;
+                    throw new Error('failed to init management');
+                }
+            });
+        });
+    }
+
+    stop() {
+        if (this.wss) {
+            this.wss.close(() => {
+                logger.info('server shutdown');
+            });
+        }
+    }
+
+    startServer() {
+        this.wss = new WebSocket.Server({
+            port: this.port,
+            clientTracking: true,
+            path: '/watch'
+        });
 
-process.env.REPORT_TOKEN = process.env.REPORT_TOKEN
-                           || _config.reportToken
-                           || Uuid.v4();
+        this.wss.on('connection', this.onConnection.bind(this));
+        this.wss.on('listening', this.onListening.bind(this));
+        this.wss.on('error', this.onError.bind(this));
 
-setTimeout(() => {
-    initManagement(logger.newRequestLogger());
-}, 5000);
+        setInterval(this.checkBrokenConnections.bind(this),
+                    CHECK_BROKEN_CONNECTIONS_FREQUENCY_MS);
 
-setTimeout(() => {
-    managementAgentWS();
-}, 6);
+        addOverlayMessageListener(this.onNewOverlay.bind(this));
+    }
 
-function managementAgentWS() {
-    const port = _config.managementAgent.port || 8010;
-    const wss = new WebSocket.Server({
-        port: port,
-        clientTracking: true,
-        path: '/watch'
-    });
+    onConnection(socket, request) {
+        function hearthbeat() {
+            this.isAlive = true;
+        }
 
-    wss.on('connection', ws => {
-        logger.info('management agent: client connected to watch route');
-    });
+        logger.info('client connected to watch route', {
+            ip: request.connection.remoteAddress,
+        });
 
-    wss.on('listening', ws => {
-        logger.info('management agent websocket server listening', { port });
-    });
+        socket.isAlive = true;
+        socket.on('pong', hearthbeat.bind(socket));
 
-    wss.on('error', error => {
-        logger.error('management agent websocket server error', { error });
-    });
+        if (!socket.readyState === socket.OPEN) {
+            logger.error('client socket not in ready state', { socket });
```

Comment:
This could include a lot of unpleasant data to look at that can be big or circular, so for clarity and for preserving performance we should select a few meaningful attributes (if available) such as `host`, `port`, `readyState` and so on